packages/components/pie-cookie-banner/src/index.ts

```diff
@@ -86,6 +95,65 @@ export class PieCookieBanner extends LitElement implements CookieBannerProps {
         this._isModalOpen = true;
     };
 
+    /**
+     * Handles the logic of the toggle switch nodes (preferences).
+     * Clicking the “all” toggle switch should turn on all preferences.
+     * When the “all” toggle is checked, and one of the other preferences is clicked,
+     * then the “all” toggle should be unchecked.
+     */
+    private _handlePreferencesToggled = (e: CustomEvent) : void => {
+        const { id } = e?.currentTarget as HTMLInputElement;
+        const turnAllNode = [...this._preferencesNodes].find((node) => node.id === 'all') as PieToggleSwitch;
+
+        if (id === turnAllNode.id) {
+            const isChecked = e.detail;
+            this._preferencesNodes.forEach((node) => {
+                node.isChecked = node.isDisabled ? node.isChecked : isChecked;
+            });
+        } else {
+            turnAllNode.isChecked ||= false;
+        }
+    };
+
+    /**
+     * Renders the content of the preference item.
+     * @private
+     */
+    private renderPreference ({
+        id, title, description, isChecked, isDisabled, hasDivider,
+    }: Preference): TemplateResult {
+        return html`
+            <div class="c-cookieBanner-preferences-item">
+                <div>
+                    <h2 class="c-cookieBanner-title">${title}</h2>
```

Comment:
If Manage your preferences is an `h2` then I think it would make more sense for these to be `h3`